src/nupic/types/BasicType.cpp

```diff
@@ -212,8 +225,19 @@ static void cpyarray(void *toPtr, const void *fromPtr, size_t count, F minVal, F
   }
 }
 
+template <typename T>
+static void cpyIntoSDR(Byte *toPtr, const T *fromPtr, size_t count) {
```

Comment:
Switch statement is now 446 lines long  :)

Reply:
> Switch statement is now 446 lines long :)

Actually this is a tree of switch statements. No one switch is more than 12 elements.
I could not think of a better way to do this. It cannot be templated because we are not switching on C++ type.  Perhaps with 12 polymorphous classes.   It may be ugly but it is readable.